File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        max_seq = 0
        for i, num in enumerate(nums):
            count = 0
            if num-1 in num_set:
                continue
            while num+count in num_set:
                count+=1

            max_seq = max(max_seq, count)

        return max_seq
            

File: Data Structures & Algorithms/max-consecutive-ones/submission-3.cpp
class Solution {
   public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int maxSequence = 0;
        int l = 0, r = 0;
        while (r < nums.size()) {
            if (nums[r] == 1) {
                maxSequence = max(maxSequence, r - l + 1);
            } else {
                l = r + 1;
            }
            r++;
        }
        return maxSequence;
    }
};